internal/configurations/configurations.go

```diff
@@ -251,34 +254,109 @@ func ForService(ctx context.Context, kubeClient *kubernetes.Cluster, service *mo
 	if err != nil {
 		return nil, err
 	}
+
 	filteredSecrets := filterSecretsByType(secretList.Items, service.SecretTypes)
 
-	return filteredSecrets, nil
+	// COMPATIBILITY SUPPORT for services from before https://github.com/epinio/epinio/issues/1704 fix
+	// Look for secrets referencing a (helm controller)-based service.
+
+	secretSelectorHC := labels.Set(map[string]string{
+		"app.kubernetes.io/instance": names.ServiceHelmChartName(service.Meta.Name, service.Meta.Namespace),
+		ConfigurationLabelKey:        "true",
+		ConfigurationTypeLabelKey:    "service",
+	}).AsSelector()
+
+	listOptionsHC := metav1.ListOptions{
+		LabelSelector: secretSelectorHC.String(),
+	}
+
+	secretListHC, err := kubeClient.Kubectl.CoreV1().Secrets(service.Meta.Namespace).List(ctx, listOptionsHC)
+	if err != nil {
+		return nil, err
+	}
+
+	filteredSecretsHC := filterSecretsByType(secretListHC.Items, service.SecretTypes)
+
+	// Merge the two lists ...
+
+	if len(filteredSecrets) == 0 {
+		return filteredSecretsHC, nil
+	}
+
+	if len(filteredSecretsHC) == 0 {
+		return filteredSecrets, nil
+	}
+
+	// A merge is actually required.
+
+	return append(filteredSecrets, filteredSecretsHC...), nil
 }
 
-// LabelServiceSecrets will look for the Opaque secrets released with a service, looking for the
-// app.kubernetes.io/instance label, then it will add the Configuration labels to "create" the configurations
-func LabelServiceSecrets(ctx context.Context, kubeClient *kubernetes.Cluster, service *models.Service) ([]v1.Secret, error) {
+// ForServiceUnlabeled returns a slice of unlabeled secrets matching the given Service
+func ForServiceUnlabeled(ctx context.Context, kubeClient *kubernetes.Cluster, service *models.Service) ([]v1.Secret, error) {
 	secretSelector := labels.Set(map[string]string{
-		"app.kubernetes.io/instance": names.ServiceHelmChartName(service.Meta.Name, service.Meta.Namespace),
+		"app.kubernetes.io/instance": names.ServiceReleaseName(service.Meta.Name),
```

Comment:
Same as before

Reply:
Done. Commit 520c57020976e9267154fb21ff968e37371155ad. Let us see what CI thinks of this.